src/Databases/OpenSearch/test/OpenSearchTests.cs

```diff
@@ -18,34 +19,166 @@ public class OpenSearchTests
+
+            var chatRequest = ChatRequest.ToChatRequest(message);
+            chatRequest.Image = image;
+
+            var response = await model.GenerateAsync(chatRequest);
+
+            var document = new Document
+            {
+                PageContent = response,
+                Metadata = new Dictionary<string, object>()
+                {
+                    {response, image}
+                }
+            };
+
+            documents.Add(document);
+        }
+
+        var pages = await _vectorStore!.AddImagesAsync(documents);
+    }
+
+    [Test]
+    public async Task can_query_image_against_images()
+    {
+        await setup_image_tests();
+      
+        var path = Path.Combine(Path.GetTempPath(), "test_image.jpg");
+        var imageData = await File.ReadAllBytesAsync(path);
+        var binaryData = new BinaryData(imageData, "image/jpg");
+
+        var embeddingRequest = new EmbeddingRequest
+        {
+            Strings = new List<string>(),
+            Images = new List<Data> { Data.FromBytes(binaryData.ToArray()) }
+        };
+        var embedding = await _embeddings.CreateEmbeddingsAsync(embeddingRequest)
+            .ConfigureAwait(false);
+
+        var floats = embedding.ToSingleArray();
+        IEnumerable<Document> similaritySearchByVectorAsync = await (_vectorStore?.SimilaritySearchByVectorAsync(floats)!).ConfigureAwait(false);
+
+        Console.WriteLine("Count: " + similaritySearchByVectorAsync.Count());
+    }
+
+    [Test]
+    public async Task can_query_text_against_images()
+    {
+        await setup_image_tests();
+
+        var llm = new Claude3SonnetModel(_provider);
+        var index = new VectorStoreIndexWrapper(_vectorStore!);
+
+        var promptText =
+            @"Use the following pieces of context to answer the question at the end. If the answer is not in context then just say that you don't know, don't try to make up an answer. Keep the answer as short as possible.
+
+{context}
+
+Question: {question}
+Helpful Answer:";
+
+        var chain =
+            Set("tell me about the orange shirt", outputKey: "question")                     // set the question
+            | RetrieveDocuments(index, inputKey: "question", outputKey: "documents", amount: 10) // take 5 most similar documents
+            | StuffDocuments(inputKey: "documents", outputKey: "context")                       // combine documents together and put them into context
+            | Template(promptText)                                                              // replace context and question in the prompt with their values
+            | LLM(llm);                                                                       // send the result to the language model
+
+        var res = await chain.Run("text");
+        Console.WriteLine(res);
     }
 
+    #endregion
+
     #region Query Simple Documents
 
+    [Test]
```

Comment:
Need to remove attribute and replace Task return type to void

Reply:
resolved